physx.js

```diff
@@ -612,6 +612,74 @@ const physxWorker = (() => {
       } : null;
     }
   };
+  w.raycastPhysicsArray = (physics, p, q, n) => {
+    if (physics && (n > 0 && n < 1000)) {
+
+
+      // Input parameters
+      for(let i=0;i<n;i++) {
+        // Position
+        p[i].toArray(scratchStack.f32, (i * 3));
+
+        // Quaternion
+        localVector.set(0, 0, -1)
+        .applyQuaternion(q[i])
+        .toArray(scratchStack.f32, (n * 3) + (i * 3));

```

Comment:
I had to stop and think about why we are doing these offsets this way.

IMO it would be more clear if before this you declare an alias for these slots, like 

```
directions = new Float32Array(scratchStack.f32.buffer, scratchStack.f32.byteOffset + (n * 3), numDirections * 3);
```

Then this line would be

```
.toArray(directions, i * 3);
```

That way each array does not need to be aware of the previous arrays, which feels messy.

Reply:
Last commit is taking care of this issue